File: codeforces/problemset/82A.py
n = int(input())
q = ['Sheldon', 'Leonard', 'Penny', 'Rajesh', 'Howard']
p = 0
cnt = 0
while True:
    steps = 0
    for i in range(1, 6):
        cnt += pow(2, p)
        if cnt >= n:
            print(q[steps])
            exit()
        steps += 1
    p += 1

File: codeforces/problemset/82A.cpp
#include <iostream>
using namespace std;
int main() {
    int n; 
    cin >> n;
    char names[100][100] = {"Sheldon", "Leonard", "Penny", "Rajesh", "Howard"};
    int cnt = 0;
    int p = 1;
    for (;;) {
        for (int i = 0; i < 5; i++) { 
            cnt += p;
            if (cnt >= n) {
                cout << names[i] << endl;
                return 0;
            }
        }
        p <<= 1;
    }
    return 0;
}
